lua/telescope/previewers.lua

```diff
@@ -46,23 +54,26 @@ local bat_maker = function(filename, lnum, start, finish)
   end
 
   return flatten {
-    command, bat_options, "--", filename
+    command, bat_options, "--", vim.fn.expand(filename)
   }
 end
 
 -- TODO: Add other options for cat to do this better
 local cat_maker = function(filename, _, _, _)
-  if vim.fn.executable('file') then
-    local handle = io.popen('file --mime-type -b ' .. filename)
-    local mime_type = vim.split(handle:read('*a'), '/')[1]
-    handle:close()
+  if get_file_stat(filename).type == 'directory' then
+    return { 'ls', '-la' }
+  end
+
+  if 1 == vim.fn.executable('file') then
+    local output = utils.get_os_command_output('file --mime-type -b ' .. filename)
+    local mime_type = vim.split(output, '/')[1]
     if mime_type ~= "text" then
```

Comment:
I don't see a way to get the `mime_type` with `vim.loop.fs_`. At least `fs_stat` will return me file for binaries and text files, and the other functions don't look promising. And i think running `file` isn't that bad. Because this code will only run if a new entry gets selected.